test/ctia/task/migration/store_test.clj

```diff
@@ -0,0 +1,304 @@
+                                                      nil)]
+        (is (= 80 (count fetched-no-query)) "limit was not properly handled")
+        (is (= 50 (count fetched-batch-1)) "query was not propertly handled on batch 1")
+        (is (= 40 (count fetched-batch-2)) "query was not propertly handled on batch 2")
+        (is (= 80 (count fetched-events-3)) "search_after was not properly handled")
+        (is (apply < (map :timestamp (concat fetched-events-1
+                                             fetched-events-2
+                                             fetched-events-3)))
+            "offset and asc orders should be properly applied, events must be sorted on timestamp")
+        (is (apply > (map :timestamp fetched-events-4))
+            "desc sort was not properly applied"))))
+
+  (testing "query-fetch-batch should properly sort entities on modified field"
+    (let [indexname "test_index"
+          tool-store {:conn es-conn
+                      :indexname indexname
+                      :mapping "tool"
+                      :type "tool"
+                      :settings {}
+                      :config {}}
+          tool-batch (map #(hash-map :id (str "tool-" %)
+                                     :timestamp (rand-int 100)
+                                     :modified %)
+                          (range 100))
+          malware-store {:conn es-conn
+                         :indexname indexname
+                         :mapping "malware"
+                         :type "malware"
+                         :settings {}
+                         :config {}}
+          malware-batch (map #(hash-map :id (str "malware-" %)
+                                        :timestamp (rand-int 100)
+                                        :modified %)
+                             (range 100))]
+      (sut/store-batch tool-store tool-batch)
+      (sut/store-batch malware-store malware-batch)
+      (es-index/refresh! es-conn indexname)
+      (let [{fetched-tool-asc :data} (sut/fetch-batch tool-store 80 0 "asc" nil)
+            {fetched-tool-desc :data} (sut/fetch-batch tool-store 80 0 "desc" nil)
+            {fetched-malware-asc :data} (sut/fetch-batch malware-store 80 0 "asc" nil)
+            {fetched-malware-desc :data} (sut/fetch-batch malware-store 80 0 "desc" nil)]
+        (is (apply < (map :modified (concat fetched-tool-asc))))
+        (is (apply > (map :modified (concat fetched-tool-desc))))
+        (is (apply < (map :modified (concat fetched-malware-asc))))
+        (is (apply > (map :modified (concat fetched-malware-desc)))))))
+
+  (es-index/delete! es-conn "test_*"))
+
+(deftest fetch-deletes-test
+  (whoami-helpers/set-whoami-response "45c1f5e3f05d0"
+                                      "foouser"
+                                      "foogroup"
+                                      "user")
+  (post-bulk examples)
+  (Thread/sleep 1000) ;; ensure indices refresh
```

Comment:
It would be safer with a loop looking at the data that cool fail after too much attempt (after 10s for example).

Reply:
It could be better to use the refresh? param instead of a Thread/sleep